planner/core/planbuilder.go

```diff
@@ -558,6 +558,15 @@ func (b *PlanBuilder) getPossibleAccessPaths(indexHints []*ast.IndexHint, tblInf
 		}
 
 		hasScanHint = true
+		if hint.IndexNames == nil {
+			// If indexNames is nil, choose TablePath.
+			// For example: `select a from t use index()`.
+			if path := getTablePath(publicPaths); path != nil {
+				hasUseOrForce = true
+				path.forced = true
```

Comment:
You need check whether is ignore or force.
Or you'll set table path as forced although no index is ignored by `IGNORE INDEX()`

Reply:
fixed